firestore-service.js

```diff
@@ -81,11 +85,13 @@ async function createDoc({ pathname }, body) {
   }
 }
 
-async function postData({ pathname, query }, body) {
+async function postData({ pathname }, body) {
   try {
+    const id = getIdFromPath(pathname);
+    const path = id ? getCollectionPath(pathname) : pathname;
```

Comment:
I think you will always have to call this two functions together. How about creating only 1 function that returns an object with `{ path, id }`